geonode/api/resourcebase_api.py

```diff
@@ -435,17 +441,57 @@ def get_list(self, request, **kwargs):
             to_be_serialized)
         return self.create_response(request, to_be_serialized)
 
-    def create_response(
-            self,
-            request,
-            data,
-            response_class=HttpResponse,
-            **response_kwargs):
-        """
-        Extracts the common "which-format/serialize/return-response" cycle.
+    def apply_perm_sorting(self, request, objs):
 
-        Mostly a useful shortcut/hook.
-        """
+        permitted_view_id = get_objects_for_user(
+            request.user,
+            'base.view_resourcebase').values_list(
+            'id',
+            flat=True)
+
+        permited_change_perms = get_objects_for_user(
+            request.user,
+            'base.change_resourcebase_permissions').values_list(
+            'id',
+            flat=True)
+
+        obj_no_perms = []
+        obj_perms = []
+
+        # retrieving in the json response the permisions
+        # to view resource 'can_view_resource'
+        # to change resource perms 'can_change_perms'
+        for idx, obj in enumerate(objs):
+            if self.get_resource_type(obj) == "map":
+                c_map = Map.objects.all().get(resourcebase_ptr_id=obj['id'])
+                obj['can_view_resource'] = is_map_viewable_by_user_utils(request.user, c_map)
+
+                # for a map we only check if user has right to change perms for the map itself
+                if obj['id'] in permited_change_perms:
+                    obj['can_change_perms'] = True
+                else:
+                    obj['can_change_perms'] = False
+
+                if obj['can_view_resource']:
```

Comment:
Actually there is no mistake here. We use can_view_resource to create the sorting, but it has no effect and impact on the permissions. 
For the permissions, we always use view_resourcebase, as in original GeoNode instance. view_resourcebase is reserved to users who can view the resource (and its data) really, whereas the can_view_resource is just used to create the sorting.